design-proposals/generated-disk-serial.md

```diff
@@ -0,0 +1,151 @@
+# Overview
+This proposal introduces a new feature in KubeVirt to automatically
+generate serial numbers for each disk attached to a virtual machine.
+
+## Motivation
+Some applications running in the VM require the disk serial number to be defined and persistent across reboots.
+Generating the serial reduces manual configuration, and in the future it can be part of a `VirtualMachinePreference` for the OS. 
+
+## Goals
+- Add an option to the VM's `.spec` that will enable automatic serial number generation for disks that
+  don't have a serial number specified.
+
+- Generate unique serial numbers for each disk. The number has to be unique compared to all other disks attached
+  to all VMs with this disk.
+ 
+- Don't store the serial number in the VM's `.spec`. It would confuse gitops operators.
+
+- The serial number should remain the same across storage migration.
+
+## Non Goals
+
+- No customization options for serial number format.
+
+## Definition of Users
+
+- VM owners
+
+## User Stories
+
+- As a VM owner, I want to use the automated disk serial number assignment to reduce manual configuration.
+- As a VM owner, I want each disk to have a unique serial number for applications that need it.
+- As a VM owner, I want to use a `VirtualMachinePreference` that is properly configured for the OS without
+  knowing about disk serial numbers.
+
+## Repos
+- [kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+A new boolean option `generateDiskSerials` will be added to the VMI `.spec`. 
+When it is set to `true`, kubevirt will generate a serial number for each disk, that does not have one specified.
+These serial numbers will then be set in the libvirt domain xml.
+
+## Persistence of generated serial numbers  
+
+The serial number has to be the same across VM restarts and storage migration,
+so it has to be stored somewhere, or generated deterministically.
+
+One possibility is to store it in an annotation on the PVC object.
+The annotation would be created when the VMI is created. 
+Then it will be read by `virt-handler` when creating libvirt domain XML.
+
+### Other persistence options
```

Comment:
For me it would be more easy to read if there would be a list of all options, with pro and cons, and highlighted in the end which option will be implemented.

Reply:
Done.